doc/contributing.rst

```diff
@@ -523,16 +427,40 @@ General commit rules
 
       #!/bin/sh
       git diff --check --cached || exit $?
-5. Describe your commit and use your common sense.
-   Example commit message::
+5. Describe your commit and use your common sense. Commit message should
+   contain enough information in order for end user to understand what is
+   going on by looking at the github log.
 
-     Fixes #123; refs #124
+   Example of bad commit message::
 
-   indicates that issue ``#123`` is completely fixed (GitHub may automatically
-   close it when the PR is committed), wheres issue ``#124`` is referenced
-   (e.g.: partially fixed) and won't close the issue when committed.
+     Fixes #123; refs #124
 
-6. PR body (not just PR title) should contain references to fixed/referenced github
+   indicates that issue ``#123`` is completely fixed (GitHub may
+   automatically close it when the PR is committed), wheres issue ``#124``
+   is referenced (e.g.: partially fixed) and won't close the issue when
+   committed. This tells reader **nothing** about what the commit did,
+   requires going into two separate github pages and reading through
+   multiple comments in order to figure out what was *wrong* - not what
+   *changed* in the commit.
+
+   Instead, commit title should contain a human-readable text (we are
+   optimizing for humans after all)
+
+   - Keep the title length to 50 characters, and commit message body to 72
+     characters.
+   - If you want to use automatic "fixes" feature of the github, please
+     supply full URL instead, it makes it much easier to copy-paste and see
+     (with pure has you first need to understand whether change was in
+     ``issues/`` or ``pull/``, then open random PR/issue and replace hash
+     URL).
+   - Consider using "Fixes" for commits that fix bugs, and "Closes" for
+     commits that implement features.
+   - Most of the points on this list were derived from `How To Write a Good
+     Commit Message
+     <https://api.coala.io/en/latest/Developers/Writing_Good_Commits.html>`_ -
```

Comment:
I think it makes sense to keep a summary of important points directly in the contribution guide instead of simply linking the external article.

Reply:
Ah ha, sounds like you're also on the excerpt train, I dig it.